etc/spack/defaults/config.yaml

```diff
@@ -14,6 +14,16 @@
 #   ~/.spack/config.yaml
 # -------------------------------------------------------------------------
 config:
+  # If 'true' enforces RPATH for ELF binaries, if 'false'
+  # enforces RUNPATH. See:
+  #
+  # http://blog.qt.io/blog/2011/10/28/rpath-and-runpath/
+  # http://xahlee.info/UnixResource_dir/_/ldpath.html
+  # https://gms.tf/ld_library_path-considered-harmful.html
+  #
+  # for more information on the subject.
+  rpath: true
```

Comment:
Let's not make this the first entry in the default `config.yaml`, as I suspect it's *not* the first thing most users want to look at.  The comment is a bit obscure -- how about:

```
# Control whether Spack embeds RPATH or RUNPATH attributes in ELF binaries
# so that they can find their dependencies. Has no effect on macOS.
# 
# There are two options:
#
#   1. rpath: use RPATH: force --disable-new-tags flag to be passed to ld
#   2. runpath: use RUNPATH: force --enable-new-tags flag to be passed to ld
#
# RPATH search paths have higher precedence than LD_LIBRARY_PATH
# and ld.so will search for libraries in transitive RPATHs of
# parent objects.
#
# RUNPATH search paths have lower precedence than LD_LIBRARRY_PATH,
# and ld.so will ONLY search for dependencies in the RUNPATH of
# the loading object.
#
# DO NOT MIX these within the same install tree. See the Spack
# documentation for details.
```

And make the two possible options `rpath` and `runpath` (enforced by schema).  This at least gives us room to choose other values in the future.

I think the links there, and maybe also https://bugs.launchpad.net/ubuntu/+source/eglibc/+bug/1253638, could be included in the Spack documentation.  Maybe some of the text above could be too... the only important parts for hte config file are really the first couple sentences, the options, and the instructions not to mix them (since if you do, RUNPATH breaks everything).

Reply:
> See the Spack documentation for details.

Please point to a specific part of the documentation

Reply:
Well it doesn't exist yet, but the idea was that he'd write it :).

Reply:
I meant that the comment should have a more specific notion on where I should look within the documentation. Having to look through all of it (or using the search function) just seems needlessly mean to the user ;)